metaflow/client/core.py

```diff
@@ -269,13 +269,19 @@ def __init__(
         _parent: Optional["MetaflowObject"] = None,
         _namespace_check: bool = True,
         _current_namespace: Optional[str] = None,
+        _current_metadata: Optional[str] = None,
     ):
         self._metaflow = Metaflow()
         self._parent = _parent
         self._path_components = None
         self._attempt = attempt
         self._current_namespace = _current_namespace or get_namespace()
+        self._current_metadata = _current_metadata or get_metadata()
         self._namespace_check = _namespace_check
+
+        # Set the metadata being passed..
+        metadata(self._current_metadata)
```

Comment:
we don't want to do this at all. Basically, we create a Metaflow object that has self.metadata in it. We want to initialize self.metadata to be the one from self._current_metadata.